api/scpca_portal/models/project.py

```diff
@@ -750,12 +752,31 @@ def get_metadata_field_names(self, columns, modality):
             ),
         )
 
-    def get_sample_libraries_mapping(self, sample_ids: List, libraries_metadata):
-        sample_libraries_id_mapping = {sample_id: set() for sample_id in sample_ids}
+    def get_sample_libraries_mapping(self, libraries_metadata):
+        """
+        Returns a dictionary which maps sample ids to a set of associated library ids.
+        """
+        # sample_libraries_id_mapping = {sample_id: set() for sample_id in sample_ids}
+        sample_libraries_id_mapping = {}
+
         for library_metadata in libraries_metadata:
-            sample_libraries_id_mapping[library_metadata["scpca_sample_id"]].add(
-                library_metadata["scpca_library_id"]
-            )
+            isMultiplexed = "demux_samples" in library_metadata
+
+            if not isMultiplexed:
```

Comment:
can we reverse the order of these cases?

Reply:
My reasoning was that the normal case is *not* multiplexed, and the exception *is* multiplexed. Personal preference - I tend to prefer the rule first, and then the exception afterwards. Does the following help at all?
```python
is_not_multiplexed = "demux_samples" not in library_metadata

if is_not_multiplexed:
  ...
```